articles/WideStrings.md

```diff
@@ -0,0 +1,264 @@
+---
+layout: default
+title: WideStrings
+abstract:
+  Some ROS2 users would like to use multi-byte characters for topic data, suitable for supporting multi-byte languages.  This article lays out the problem, and explores potential solutions to this problem.
+author: '[Chris Lalancette](https://github.com/clalancette)'
+published: true
+---
+
+- This will become a table of contents (this text will be scraped).
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Background
+
+ROS2 currently supports the concept of strings, which are sequences of single-byte
+characters.  When working with ROS 2.0 topic data, some users would like to use multi-byte
+characters (e.g. UTF-8) for their topic data.  This article explores the
+problem and the possible solutions.  Note that this article specifically does
+not talk about binary data, as that is already handled by using uint8_t arrays.
+It also does not talk about using multi-byte characters for the topic *names*,
+as this is disallowed by the DDS specification.
+
+## Multi-byte character background
+
+Before delving into ROS 2.0 specifics, some time should be spent discussing multi-byte
+characters in general.  What is meant by multi-byte characters?  The original ASCII
+character set only provided for printable characters using binary sequences 0 to 127.
+Several extensions were made to this to use 128-255, but that approach only extended the
+character set to be able to cover a few more languages.  Several other encodings were
+created as the need arose, and it became clear by the 1990's that a worldwide standard
+was required.  Thus, Unicode was created to be a single specification that could cover
+all of the languages of the world.  The early versions of Unicode actually failed to
+live up to this expectation, and some of those early versions of Unicode are still
+baked into widely used software (Microsoft Windows, Java, etc).  Later on, the Unicode
+standard evolved to truly be able to cover the languages of the entire world, and this
+is what should generally be used for new designs.  As a side benefit, the newer versions of
+Unicode (UTF-8) are backwards compatible with ASCII.  There is a lot of history here, and
```

Comment:
It depends on your definition of "newer"; the original Unicode (that came out in 1996 or whatever) didn't have UTF-8, which is why Win32 and Java both use UTF-16 (really UCS-2).  That being said, UTF-8 has been around a long time, so I'll change the wording here.